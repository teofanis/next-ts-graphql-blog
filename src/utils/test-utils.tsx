import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
