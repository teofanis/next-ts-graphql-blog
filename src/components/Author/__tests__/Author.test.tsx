import { Author } from '@components/index'
import { render, within } from '@utils/test-utils'
import React from 'react'
import { author } from './__fixtures__/author.json'

describe('Author Test', () => {
  it('renders correctly', () => {
    const { container } = render(<Author author={author} />)
    expect(container).toMatchSnapshot()
  })
  it('displays author details', () => {
    const { getByTestId } = render(<Author author={author} />)
    const authorDetails = getByTestId('author-details')
    expect(authorDetails).toBeInTheDocument()
    expect(authorDetails).toBeVisible()
  })

  it('displays the author name and bio', () => {
    const { getByTestId } = render(<Author author={author} />)
    const authorDetails = getByTestId('author-details')
    expect(authorDetails).toHaveTextContent(author.name)
    expect(authorDetails).toHaveTextContent(author.bio)
  })

  it('displays the author image', () => {
    const { getByTestId } = render(<Author author={author} />)
    const authorDetails = getByTestId('author-details')
    const image = within(authorDetails).getByRole('img')
    expect(image).toBeInTheDocument()
    expect(image).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
  })
})
