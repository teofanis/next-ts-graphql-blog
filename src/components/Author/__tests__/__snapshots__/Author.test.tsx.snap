// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Test renders correctly 1`] = `
<div>
  <div
    class="text-center mt-20 mb-8 p-12 relative rounded-lg bg-black bg-opacity-20"
    data-testid="author-details"
  >
    <div
      class="absolute left-0 right-2 -top-14"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Test Author Name"
          class="align-middle rounded-full"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <h3
      class="text-white my-4 text-xl font-bold"
    >
      Test Author Name
    </h3>
    <p
      class="text-white text-lg"
    >
      Test Author Bio
    </p>
  </div>
</div>
`;
